src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,320 @@
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("Click help to learn about the migration of pre-3.6 databases.")
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("However, a new database was created alongside the pre-3.6 one."),
+                                                                                      ButtonType.OK, openHelp);
+
+            result.filter(btn -> btn.equals(openHelp)).ifPresent(btn -> HelpAction.openHelpPage(HelpFile.SQL_DATABASE_MIGRATION));
+            result.filter(btn -> btn == ButtonType.OK).ifPresent(btn -> openSharedDatabase());
+
+        }
+
+        setLoadingConnectButtonText(false);
+    }
+
+    private void setLoadingConnectButtonText(boolean isLoading) {
+        btnConnect.setDisable(isLoading);
+        if (isLoading) {
+            btnConnect.setText(Localization.lang("Connecting..."));
+        } else {
+            btnConnect.setText(Localization.lang("Connect"));
+        }
+    }
+
+    private boolean isSharedDatabaseAlreadyPresent() {
+        List<BasePanel> panels = frame.getBasePanelList();
+        return panels.parallelStream().anyMatch(panel -> {
+            BibDatabaseContext context = panel.getBibDatabaseContext();
+
+            return ((context.getLocation() == DatabaseLocation.SHARED) &&
+                    this.connectionProperties.equals(context.getDBMSSynchronizer().getConnectionProperties()));
+        });
+    }
+
+    private void setPreferences() {
+        prefs.setType(cmbDbType.getSelectionModel().getSelectedItem().toString());
+        prefs.setHost(tbHost.getText());
+        prefs.setPort(tbPort.getText());
+        prefs.setName(tbDb.getText());
+        prefs.setUser(tbUser.getText());
+
+        if (chkRememberPassword.isSelected()) {
+            try {
+                prefs.setPassword(new Password(tbPwd.getText(), tbPwd.getText()).encrypt());
+            } catch (GeneralSecurityException | UnsupportedEncodingException e) {
+                LOGGER.error("Could not store the password due to encryption problems.", e);
+            }
+        } else {
+            prefs.clearPassword(); // for the case that the password is already set
+        }
+
+        prefs.setRememberPassword(chkRememberPassword.isSelected());
+    }
+
+    private void checkFields() throws JabRefException {
+        if (isEmptyField(tbHost)) {
```

Comment:
Validation should be done using the validation framework so that the user get immediate feedback (have a look at the usage of ControlsFxVisualizer).